### ISSUE TITLE
Implement Dihedral Checking

### ISSUE THREAD

**Issue by yudongqiu** [lpwgroup/torsiondrive#49]
This issue serves as a feature request and tracker of an improvement to the torsiondrive package.

As mentioned in #23, it will be nice to check if the dihedral specified contains a straight angle.

I think this is a good project for @hyejang to try to contribute. Here is my idea on the implementation:

Currently `torsiondrive` uses `molecule.measure_dihedrals()` function to measure the dihedral angle. We can write a new function to replace this function, which contains some nice features that are more suitable for `torsiondrive`.

The new function should look like this:
```
def measure_dihedrals(molecule, dihedral_list, check_linear=True, check_bonded=True):
```

This function should compute the values of a list of dihedrals for the molecule. Some specs:

`molecule`: a `geometric.molecule.Molecule` object. Geometry is stored as `molecule.xyz[0]`, bonds is stored as `molecule.bonds` if available, 

`dihedral_list`: a list like `[ [i0, j0, k0, l0], [i1, j1, k1, l1] ]` which represents a list of dihedrals by atom indices.

`check_linear`: If true, check the angle of `i-j-k` and `j-k-l` of each dihedral, raise a warning message if the angle > 165 degree.

`check_bonded`: If true, check `i-j, j-k, k-l` are all bonded. If bonds are not availble, call m.build_bonds() to build them. If any bonds are missing, raise a warning.

This function should return an numpy array of dihedral values, the length is same as the `dihedral_list`.

Check points:

- [x] implement `measure_dihedrals()` function in `dihedral_scanner.py`

- [x] write unit tests for this function in `test_dihedral_scanner.py`. Cover the warnings.

- [x] replace all `molecule.measure_dihedrals` by the new function.

- [x] make sure all tests pass.

**Comment by yudongqiu** [comment]
Here is a link to the doc of the warnings module: https://docs.python.org/3/library/warnings.html#module-warnings

**Comment by yudongqiu** [comment]
To use pytest to test the warnings: https://docs.pytest.org/en/latest/warnings.html

**Comment by yudongqiu** [comment]
I just come up with a new edge case, that the input `dihedral_list` may not be valid. This could happen when dihedrals in the input list do not have 4 integers, or when the atom indices are out of bound. The ValueError should print the input dihedral_list in the error message.

**Comment by yudongqiu** [comment]
This is implemented by @hyejang in PR #52. Nice work!
